### PR TITLE
rework of the tls section

### PR DESCRIPTION
**Pull request by jsalowey**
Reworked the TLS section a little.  mostly addresses issue #46

### CODE REVIEW

File: draft-ietf-wimse-s2s-protocol.md [hunk 1]
```diff
@@ -486,25 +486,21 @@ A signed response would be:
...
 
 Servers wishing to use the WIMSE certificate for authorizing the client MUST require client certificate authentication in the TLS handshake. Other methods of post handshake authentication are not specified by this document.
 
-WIMSE clients and servers MUST validate that the trust domain portion of the WIMSE certificate matches the expected trust domain for the other side of the connection.
+WIMSE server certificates MUST have the id-kp-serverAuth extended key usage {{!RFC5280}} field set and WIMSE client certificates MUST have the id-kp-clientAuth extended key usage field set. A certificate that is used for both client and server connections may has both fields set. This specification does not make any other requirements beyond {{!RFC5280}} on the contents of WIMSE certificates or on the certificates that issue WIMSE certificates.
```
Comment (yaronf) [inline draft-ietf-wimse-s2s-protocol.md:495]: Asking out of ignorance: in the past, EKU was hit-and-miss. Is it now implemented so widely that we can have it as a MUST?

Comment (jsalowey) [reply draft-ietf-wimse-s2s-protocol.md:495]: I think its probably still a little hit and miss.  I think we can try to raise the bar here, but maybe its not really all that important enough to make it a barrier. I'd be happy with a SHOULD.

File: draft-ietf-wimse-s2s-protocol.md [hunk 2]
```diff
@@ -486,25 +486,21 @@ A signed response would be:
 
 # Using Mutual TLS for Service To Service Authentication {#mutual-tls}
 
-The WIMSE workload identity may be carried within an X.509 certificate. The WIMSE workload identity MUST be encoded in a SubjectAltName extension of type URI.  There MUST be only one SubjectAltName extension of type URI in a WIMSE certificate.  The WIMSE certificate may contain SubjectAltName extensions of other types such as DNSName.
+The WIMSE workload identity may be carried within an X.509 certificate. The WIMSE workload identity MUST be encoded in a SubjectAltName extension of type URI.  There MUST be only one SubjectAltName extension of type URI in a WIMSE certificate.  If the workload will act as a TLS server for clients that do not understand WIMSE workload identities it is RECOMMENDED that WIMSE certificate contain a SubjectAltName of DNSName with the appropriate DNS names for the server. The certificate may contain SubjectAltName extensions of other types.
```
Comment (yaronf) [inline draft-ietf-wimse-s2s-protocol.md:489]:
```suggestion
The WIMSE workload identity may be carried within an X.509 certificate. The WIMSE workload identity MUST be encoded in a SubjectAltName extension of type URI.  There MUST be only one SubjectAltName extension of type URI in a WIMSE certificate.  If the workload will act as a TLS server for clients that do not understand WIMSE workload identities it is RECOMMENDED that WIMSE certificate contain a SubjectAltName of type DNSName with the appropriate DNS names for the server. The certificate may contain SubjectAltName extensions of other types.
```

File: draft-ietf-wimse-s2s-protocol.md (hunk 1 above)
Comment (yaronf) [inline draft-ietf-wimse-s2s-protocol.md:495]:
```suggestion
WIMSE server certificates SHOULD have the id-kp-serverAuth extended key usage {{!RFC5280}} field set and WIMSE client certificates SHOULD have the id-kp-clientAuth extended key usage field set. A certificate that is used for both client and server connections may have both fields set. This specification does not make any other requirements beyond {{!RFC5280}} on the contents of WIMSE certificates or on the certification authorities that issue WIMSE certificates.
```

File: draft-ietf-wimse-s2s-protocol.md [hunk 3]
```diff
@@ -486,25 +486,21 @@ A signed response would be:
...
 
-## Host Name Validation
+## Server Name Validation {#server-name}
 
-<cref>TODO: need to define trust anchor used to validate the certificate is appropriate for the trust domain.</cref>
+If the WIMSE client uses a hostname to connect to the server and the server certificate contain a DNS SAN the client MUST perform standard host name validation ({{Section 6.3 of RFC9525}}) unless it is configured with the information necessary to validate the peer's WIMSE identity. If the client did not perform standard host name validation then WIMSE client SHOULD further use the WIMSE workload identifier to validate the server.  The host portion of the WIMSE URI is NOT treated as a host name as specified in section 6.4 of {{!RFC9525}} but rather as a trust domain. The server identity is encoded in the path portion of the WIMSE workload identifier in a deployment specific way. Validating the WIMSE workload identity could be a simple match on the trust domain and path portions of the identifier or validation may be based on the specific details on how the identifier is constructed. The path portion of the WIMSE identifier MUST always be considered in the scope of the trust domain.
```
Comment (yaronf) [inline draft-ietf-wimse-s2s-protocol.md:499]:
```suggestion
If the WIMSE client uses a hostname to connect to the server and the server certificate contain a DNS SAN the client MUST perform standard host name validation ({{Section 6.3 of RFC9525}}) unless it is configured with the information necessary to validate the peer's WIMSE identity. If the client did not perform standard host name validation then the WIMSE client SHOULD further use the WIMSE workload identifier to validate the server.  The host portion of the WIMSE URI is NOT treated as a host name as specified in section 6.4 of {{!RFC9525}} but rather as a trust domain. The server identity is encoded in the path portion of the WIMSE workload identifier in a deployment specific way. Validating the WIMSE workload identity could be a simple match on the trust domain and path portions of the identifier or validation may be based on the specific details on how the identifier is constructed. The path portion of the WIMSE identifier MUST always be considered in the scope of the trust domain.
```

**Review by yaronf — CHANGES_REQUESTED** [review]
Mostly editorial changes, but also MUST->SHOULD

File: draft-ietf-wimse-s2s-protocol.md (hunk 1 above)
Comment (jsalowey) [reply draft-ietf-wimse-s2s-protocol.md]: Resolved by Yaron's suggestion.

**Pull request merged**